payments/vipps/seamless-view.md

```diff
@@ -18,125 +18,427 @@ sidebar:
 ![Vipps_flow_PaymentPages.png]
 
-### Payment Url
+## Purchase flow
 
-{% include payment-url.md when="selecting Vipps as payment instrument" %}
+The sequence diagram below shows the two requests you have to send to
+Swedbank Pay to make a purchase.
+The links will take you directly to the API description for the specific
+request.
 
-## Screenshots
+```mermaid
+sequenceDiagram
+    Browser->>Merchant: start purchase (pay with VIPPS)
+    activate Merchant
+    Merchant->>-SwedbankPay.FrontEnd: POST /psp/vipps/payments ①
+    activate SwedbankPay.FrontEnd
+    note left of Merchant: First API request
+    SwedbankPay.FrontEnd-->>-Merchant: rel: view-payment ②
+    activate Merchant
+    Merchant-->>-Browser: authorization page
+    activate Browser
+    note left of Browser: Open iframe ③
+    Browser->>Browser: Enter mobile number ④
+    Browser-->>-SwedbankPay.FrontEnd: Passing data for authorization
+    activate SwedbankPay.FrontEnd
 
-You redirect the payer to Swedbank Pay hosted payment page to collect the
-consumers mobile number.
+    SwedbankPay.FrontEnd-->>-Vipps.API: POST <paymentId>/authorizations ⑤
+    activate Vipps.API
+    Vipps.API-->>-SwedbankPay.FrontEnd: response
+    activate SwedbankPay.FrontEnd
+    SwedbankPay.FrontEnd-->>-Browser: Authorization response (State=Pending) ⑥
+    activate Browser
+    note left of Browser: check your phone
 
-![Vipps mobile Payments]
-[Vipps-screenshot-1]{:width="426px" :height="632px"}
-![Vipps Payments][Vipps-screenshot-2]{:width="427px" :height="694px"}
+    Vipps.API-->>Vipps_App: Confirm Payment UI
+    activate Vipps_App
+    note left of Vipps.API: Dialogue with Vipps ⑦
+    Vipps_App-->>Vipps_App: Confirmation Dialogue
+    Vipps_App-->>-Vipps.API: Confirmation
+    activate Vipps.API
+    Vipps.API-->>-SwedbankPay.BackEnd: make payment
+    activate SwedbankPay.BackEnd
+    SwedbankPay.BackEnd-->>SwedbankPay.BackEnd: execute payment
+    SwedbankPay.BackEnd-->>Vipps.API: response
+    SwedbankPay.BackEnd-->>-SwedbankPay.FrontEnd: authorize result
+    activate SwedbankPay.FrontEnd
+    SwedbankPay.FrontEnd-->>-Browser: Display authorize result
+```
+
+### Explainations
+
+* ① When the payer starts the purchase process, you make a `POST` request
+  towards Swedbank Pay with the collected Purchase information.
+* ② `rel: view-payment` is a value in one of the operations, sent as a response
+  from Swedbank Pay to the Merchant.
+* ③ `Open iframe` creates the Swedbank Pay hosted iframe.
+* ④ The consumer UI page displays the payment window as content inside of the
+  `iframe`. The consumer can insert mobile information for authorization.
+* ⑤ A `POST` request is sent to the Vipps API with the mobile number for
+  authorization.
+* ⑥ The response will contain the state of the transaction. It will normally be
+  in `AwaitingActivity` in this fase of the payment.
+* ⑦ Swedbank Pay handles the dialogue with Vipps and the consumer confirms the
+  purchase in the Vipps app.
+
+### 3-D Secure
```

Comment:
I'm not sure it is relevant to say anything about 3-D Secure on Vipps payments, as it won't affect the flow of the payment process. I don't think you will ever have to go through BankID once you are up and running with Vipps. Being up and running with Vipps is a prerequisite to pay with Vipps, so that's not something we have to think about.